src/components/DiagramFrame/DiagramFrame.tsx

```diff
@@ -104,25 +104,13 @@ export const DiagramFrame: React.FC<DiagramFrameProps> = ({
 
   const onlyDiagramSelected = () => { return showNoAside && !showViewOptions && !showSettings && !showHelp ? true : undefined }
 
-  let modelDetails = unfilteredModels ? unfilteredModels.filter((d: ILookmlModel)=>{ 
-    return d.explores.length >= 1
-  }).map((d: ILookmlModel)=>{
-    return {
-      value: d.name,
-      label: d.label
-    }
-  }).sort((a: SelectOptionProps, b: SelectOptionProps) => a.label < b.label ? -1 : 1) : []
+  let modelDetails = prepareModelDropdown(unfilteredModels)
 
   let currentExplore: ILookmlModelExplore = modelDetail?.explores.filter((d: ILookmlModelExplore)=>{
     return d.name === pathExploreName
   })[0]
 
-  let exploreList: ExploreDropdown[] = currentModel?.explores.map((d: ILookmlModelExplore)=>{
-    return {
-      value: d.name,
-      label: d.label
-    }
-  }).sort((a: ExploreDropdown, b: ExploreDropdown) => a.label < b.label ? -1 : 1)
+  let exploreList: ExploreDropdown[] = prepareExploreList(currentModel)
```

Comment:
const

Reply:
There are actually a number of places where you are using let which could be const

Reply:
Yeah, good call. Will resolve the relevant ones here but will continue to fix this over next PRs